tests/filecheck/dialects/riscv/riscv_ops_custom_syntax.mlir

```diff
@@ -0,0 +1,270 @@
+  // CHECK-NEXT: %csrrci_r = riscv.csrrci 1024, 0 : () -> (!riscv.reg<>)
+  // CHECK-NEXT: %csrrwi_rw = riscv.csrrwi 1024, 1 : () -> (!riscv.reg<>)
+  // CHECK-NEXT: %csrrwi_w = riscv.csrrwi 1024, w, 1 : () -> (!riscv.reg<>)
+
+  // Machine Mode Privileged Instructions
+  "riscv.wfi"() : () -> ()
+  // CHECK-NEXT: riscv.wfi : () -> ()
+
+
+  // RV32M/RV64M: 7 “M” Standard Extension for Integer Multiplication and Division
+
+  // Multiplication Operations
+  %mul = "riscv.mul"(%0, %1) : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+  %mulh = "riscv.mulh"(%0, %1) : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+  %mulhsu = "riscv.mulhsu"(%0, %1) : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+  %mulhu = "riscv.mulhu"(%0, %1) : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+  // CHECK-NEXT: %mul = riscv.mul %0, %1 : (!riscv.reg<>, !riscv.reg<>) -> (!riscv.reg<>)
+  // CHECK-NEXT: %mulh = riscv.mulh %0, %1 : (!riscv.reg<>, !riscv.reg<>) -> (!riscv.reg<>)
+  // CHECK-NEXT: %mulhsu = riscv.mulhsu %0, %1 : (!riscv.reg<>, !riscv.reg<>) -> (!riscv.reg<>)
+  // CHECK-NEXT: %mulhu = riscv.mulhu %0, %1 : (!riscv.reg<>, !riscv.reg<>) -> (!riscv.reg<>)
+
+  // Division Operations
+  %div = "riscv.div"(%0, %1) : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+  %divu = "riscv.divu"(%0, %1) : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+  %rem = "riscv.rem"(%0, %1) : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+  %remu = "riscv.remu"(%0, %1) : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+  // CHECK-NEXT: %div = riscv.div %0, %1 : (!riscv.reg<>, !riscv.reg<>) -> (!riscv.reg<>)
+  // CHECK-NEXT: %divu = riscv.divu %0, %1 : (!riscv.reg<>, !riscv.reg<>) -> (!riscv.reg<>)
+  // CHECK-NEXT: %rem = riscv.rem %0, %1 : (!riscv.reg<>, !riscv.reg<>) -> (!riscv.reg<>)
+  // CHECK-NEXT: %remu = riscv.remu %0, %1 : (!riscv.reg<>, !riscv.reg<>) -> (!riscv.reg<>)
+
+  // Assembler pseudo-instructions
+
+  %li = "riscv.li"() {"immediate" = 1 : i32}: () -> !riscv.reg<>
+  // CHECK-NEXT: %li = riscv.li 1 : () -> (!riscv.reg<>)
+
+  // Environment Call and Breakpoints
+  "riscv.ecall"() : () -> ()
+  "riscv.ebreak"() : () -> ()
+  "riscv.directive"() {"directive" = ".align", "value" = "2"} : () -> ()
+  "riscv.directive"() ({
+    %nested_li = "riscv.li"() {"immediate" = 1 : i32} : () -> !riscv.reg<>
+  }) {"directive" = ".text"} : () -> ()
+  // CHECK-NEXT: riscv.ecall : () -> ()
+  // CHECK-NEXT: riscv.ebreak : () -> ()
+  // CHECK-NEXT: riscv.directive ".align", "2" : () -> ()
+  // CHECK-NEXT: riscv.directive ".text" ({
+  // CHECK-NEXT:   %nested_li = riscv.li 1 : () -> (!riscv.reg<>)
+  // CHECK-NEXT: }) : () -> ()
+
+  // Custom instruction
+  %custom0, %custom1 = "riscv.custom_assembly_instruction"(%0, %1) {"instruction_name" = "hello"} : (!riscv.reg<>, !riscv.reg<>) -> (!riscv.reg<>, !riscv.reg<>)
+  // CHECK-NEXT: %custom0, %custom1 = riscv.custom_assembly_instruction %0, %1, "hello" : (!riscv.reg<>, !riscv.reg<>) -> (!riscv.reg<>, !riscv.reg<>)
```

Comment:
This is misleading, the function name is hello, it's not an argument. I'd expect something like this:
```
%custm0, %custom1 = riscv.custom_assembly_instruction hello (%0 : reg<>), (%1 : reg<>)
```

Reply:
Or I guess like this with my other suggestions, you're in the best position to compare them locally with the different formats
```
%custm0, %custom1 = riscv.custom_assembly_instruction hello %0, %1 : (x?, x?) -> (x?, x?)
```

Reply:
Done